tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/DeLambdifier.scala

```diff
@@ -0,0 +1,29 @@
+package at.forsyte.apalache.tla.pp
+
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.oper.TlaOper
+import at.forsyte.apalache.tla.lir.transformations._
+
+/**
+ * DeLambdifier turns applications of operator-typed LET-expressions, into LET-expressions with applications pushed into
+ * the body.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+class DeLambdifier(tracker: TransformationTracker) extends TlaExTransformation {
```

Comment:
I think `LetInApplier` is probably a more accurate name? There seems to already be precedent for `de-lambda` as a transformation that converts (proper) lambdas into let/in expressions, which is not what we're ding here :) https://en.wikipedia.org/wiki/Let_expression#Conversion_from_lambda_to_let_expressions